test/unit/registry-routes-component.js

```diff
@@ -465,6 +465,53 @@ describe('registry : routes : component', function(){
     });
   });
 
+  describe('when getting a component with server.js that sets custom headers with empty customHeadersToSkipOnWeakVersion', function() {
+    var code, response, headers;
+
+    before(function(done) {
+      initialise(mockedComponents['response-headers-component']);
+      componentRoute = new ComponentRoute({}, mockedRepository);
+
+      var resJson = function(calledCode, calledResponse) {
+        code = calledCode;
+        response = calledResponse;
+        done();
+      };
+
+      var resSet = function(calledHeaders) {
+        headers = calledHeaders;
+      };
+
+      componentRoute({
+        headers: {},
+        params: { componentName: 'response-headers-component', componentVersion: '1.X.X' }
+      }, {
+        conf: {
+          baseUrl: 'http://component.com/',
+          executionTimeout: 0.1,
+          customHeadersToSkipOnWeakVersion: ['test-header']
+        },
+        json: resJson,
+        set: resSet
+      });
+    });
+
+    it('should return 200 status code', function() {
+      expect(code).to.be.equal(200);
+    });
+
+    it('should not set the HTTP response test-headers', function() {
+      expect(response.headers).to.not.be.null;      
+      expect(response.headers['test-header']).to.equal('test-value');
+      expect(headers).to.be.undefined;      
```

Comment:
I think I can't see any value for having `headers` into the GET /component response body, as we should use the headers for that. So, for simplicity, I agree about stripping the headers from the headers :rofl: in case this setting is on, and I vote for just removing the `headers` from the response in any case.

In case of the POST batch, I think it is a bit different. Using the response body for the headers is the simplest way to include the headers in each component. The way I would tackle it there, is to have a `headers` key in the response that behaves exactly as the headers in the GET. Still, I think we can do the POST in a separate PR if you agree.